sdk/dapp-kit/src/components/SuiClientProvider.tsx

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { SuiClient, getFullnodeUrl } from '@mysten/sui.js/client';
+import { SuiClientContext } from '../hooks/useSuiClient.js';
+import { useMemo } from 'react';
+
+export interface SuiClientProviderProps {
+	children: React.ReactNode;
+	client?: SuiClient;
+	url?: string;
+}
+
+export const SuiClientProvider = (props: SuiClientProviderProps) => {
+	const client = useMemo(
+		() =>
+			props.client ??
+			new SuiClient({
+				url: props.url ?? getFullnodeUrl('devnet'),
+			}),
+		[props.client, props.url],
+	);
+
+	return <SuiClientContext.Provider value={client}>{props.children}</SuiClientContext.Provider>;
```

Comment:
I don't think this context provider should actually expose the client directly, ideally it would expose an object with the client, and methods for manipulating the context (eg, switching client/network could be part of the context api).  This can be done later, this isn't really about creating the final dapp-kitt API yet